### PR TITLE
Implement Create/Destroy SnapshotStep

### PR DESCRIPTION
**Pull request by mkpankov**
Signed-off-by: Michael Pankov <work@michaelpankov.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2140)

### CODE REVIEW

File: chroma_core/models/host.py [hunk 1]
```diff
@@ -1785,6 +1785,20 @@ class NoNidsPresent(Exception):
     pass
 
 
+class CreateSnapshotStep(Step):
+    def run(self, kwargs):
+        self.invoke_rust_agent_expect_result(
+            kwargs["host"], "snapshot_create", [kwargs["fsname"], kwargs["name"], kwargs["comment"]]
```
Comment (jgrund) [inline chroma_core/models/host.py:1791]: The value being sent needs to be: 

https://github.com/whamcloud/integrated-manager-for-lustre/blob/2f5c61b761f26a23a9848c6cf2e15e10bff4c276/iml-wire-types/src/snapshot.rs#L13-L23

File: chroma_core/models/host.py [hunk 2]
```diff
@@ -1785,6 +1785,20 @@ class NoNidsPresent(Exception):
...
+        )
+
+
+class DestroySnapshotStep(Step):
+    def run(self, kwargs):
+        self.invoke_rust_agent_expect_result(
+            kwargs["host"], "snapshot_destroy", [kwargs["fsname"], kwargs["name"], kwargs["force"]]
```
Comment (jgrund) [inline chroma_core/models/host.py:1798]: The value being sent needs to be:

https://github.com/whamcloud/integrated-manager-for-lustre/blob/2f5c61b761f26a23a9848c6cf2e15e10bff4c276/iml-wire-types/src/snapshot.rs#L27-L36

File: chroma_core/models/host.py [hunk 3]
```diff
@@ -1785,6 +1785,20 @@ class NoNidsPresent(Exception):
...
+        )
+
+
 class MountSnapshotStep(Step):
     def run(self, kwargs):
         self.invoke_rust_agent_expect_result(kwargs["host"], "snapshot_mount", [kwargs["fsname"], kwargs["name"]])
```
Comment (jgrund) [inline chroma_core/models/host.py:1804]: https://github.com/whamcloud/integrated-manager-for-lustre/blob/2f5c61b761f26a23a9848c6cf2e15e10bff4c276/iml-wire-types/src/snapshot.rs#L41-L46

File: chroma_core/models/host.py (hunk 1 above)
Comment (mkpankov) [reply chroma_core/models/host.py]: Right, fixed

File: chroma_core/models/host.py [hunk 4]
```diff
@@ -1785,11 +1785,33 @@ class NoNidsPresent(Exception):
     pass
 
 
+class CreateSnapshotStep(Step):
+    def run(self, kwargs):
+        self.invoke_rust_agent_expect_result(
+            kwargs["host"],
+            "snapshot_create",
+            {"fsname": kwargs["fsname"], "name": kwargs["name"], "comment": kwargs["comment"]},
```
Comment (ip1981) [inline chroma_core/models/host.py:1793]: will it fail if comment is not defined?

Comment (mkpankov) [reply chroma_core/models/host.py:1793]: I guess it will, but our arguments structure doesn't support not having a comment.

Comment (ip1981) [reply chroma_core/models/host.py:1793]: It's Option though https://github.com/whamcloud/integrated-manager-for-lustre/commit/4c0b714950f5de892ae7beb4da9dc7c34df6a9a0#diff-da0597742d3aa1093abb8b957bb71a81R21

Comment (jgrund) [reply chroma_core/models/host.py:1793]: You could check if `kwargs`  has a "comment" key and add it to the dictionary if so.

Comment (mkpankov) [reply chroma_core/models/host.py:1793]: Ah, must've mistaken for something else.

Fixed.

**Pull request merged**